Automattic/wp-calypso
# Rewind: add flow in AL to switch from VP/activate Rewind/migrate credentials

Fixes #21850

Introduces a dialog that prompts user to switch from VP to Rewind or introduce credentials.

<img width="816" alt="captura de pantalla 2018-01-26 a la s 19 33 14" src="https://user-images.githubusercontent.com/1041600/35463528-d29ca606-02cf-11e8-948c-3e47d1d1915d.png">

### Dismiss

When users click the "No thanks" link, they're taken back to where they were in WP Admin. For this, the query parameter `rewind-redirect` is needed. It points to a page in WP Admin like this `rewind-redirect=admin.php?page=somepage`.

### Testing

When you arrive to AL with a URL like:
```
http://calypso.localhost:3000/stats/activity/sleepy-spectacled.jurassic.ninja?rewind-redirect=admin.php%3Fpage%3Djetpack
```
- verify you get the dialog
- verify that the "No thanks" link takes you back to `admin.php?page=whatever`

Depending on the site setup there are three particular flows:

a. Create a site with Jetpack and VaultPress plugins: it must have a Jetpack plan with backups and VaultPress be active. It must not have a VP plan. 
- verify you get the **Switch now** button in the dialog
- in the flow, you'll get a button to switch the site from VP to Rewind.

b. Create a site that has a Jetpack plan with backups but never had VP installed.
- verify you get the **Continue setup** button in the dialog
- in the flow, you'll get the form to add the credentials.

c. Create a site with a Jetpack free plan. Purchase a plan with credits.
- when you're in the orange **Thank You** screen, verify you have a **Setup backups** button
- click the button and verify you're put into the flow to add the credentials.
